tools/shell/shell_runner.cpp

```diff
@@ -37,11 +40,27 @@ int main(int argc, char* argv[]) {
     if (readOnlyMode) {
         systemConfig.readOnly = true;
     }
+    if (!pathToHistory.empty() && pathToHistory.back() != '/') {
+        pathToHistory += '/';
+    }
+    pathToHistory += "history.txt";
+    FILE* fp = fopen(pathToHistory.c_str(), "r");
```

Comment:
I think fopen is not cross platform, you may have to use our filesystem class which provides basic open/close functionality.